hera_cal/reflections.py

```diff
@@ -51,169 +72,80 @@
-            edgecut : int, number of channels to exclude in CLEAN on either side of band.
-                Note this is not the same as flagging edge channels. This excludes them
-                entirely, such that if a taper is applied, it is not discontinuous at
-                the band edges.
-            taper : str, Tapering function to apply across freq before FFT
-            alpha : float, if taper is Tukey, this its alpha parameter
-            timeavg : bool, if True, average data across time weighted by flags
-            broadcast_flags : bool, if True, broadcast flags across time using time_thresh
-            time_thresh : float, if fraction of flagged times exceeds this ratio,
-                flag a channel for all times.
-        """
-        # initialize containers
-        if hasattr(self, "clean_data") and not overwrite:
-            raise ValueError("self.clean_data exists and overwrite is False...")
-        self.clean_data = DataContainer({})
-        self.resid_data = DataContainer({})
-        self.clean_info = {}
-        self.clean_freqs = self.freqs
-        if edgecut > 0:
-            self.clean_freqs = self.clean_freqs[edgecut:-edgecut]
-
-        # get keys
-        if keys is None:
-            keys = self.data.keys()
-
-        # iterate over keys
-        for k in keys:
-            echo("...Cleaning data key {}".format(k), verbose=verbose)
-            (model, flag, residual, dlys,
-             info) = reflections_delay_filter(self.data[k].copy(), self.flags[k].copy(), self.dnu, tol=tol,
-                                              maxiter=maxiter, gain=gain, skip_wgt=skip_wgt, dly_cut=dly_cut,
-                                              edgecut=edgecut, taper=taper, alpha=alpha, timeavg=timeavg,
-                                              broadcast_flags=broadcast_flags, time_thresh=time_thresh)
-            # add residual back into model
-            model += residual * ~flag
-
-            # append to containers
-            self.clean_data[k] = model
-            self.resid_data[k] = residual
-            self.clean_info[k] = info
-
-        self.clean_dlys = dlys
-        self.clean_times = self.times
-        if timeavg:
-            self.clean_times = np.mean(self.clean_times, keepdims=True)
-
-    def model_reflections(self, dly_range, keys=None, data='clean', edgecut=0, taper='none',
-                          alpha=0.1, zero_pad=0, overwrite=False, fthin=10, verbose=True):
-        """
-        Model reflections in (ideally RFI-free) visibility data. To CLEAN data of
-        flags see the self.dly_clean_data() function. Recommended to set zero_pad
-        to at least as large as Nfreqs.
+        To CLEAN data of RFI gaps see the self.vis_clean() function.
+        Recommended to set zeropad to at least as large as Nfreqs/2.
 
         Args:
+            data : DataContainer, data to find reflections in.
```

Comment:
Do you want to make this an optional parameter and use self.data unless otherwise overwritten like you did with FRFilter?